2021-10-13 00:25:20,096 PyTorch Version: 1.7.1
2021-10-13 00:25:20,121 GPU List: ['Tesla V100-SXM2-32GB']
2021-10-13 00:25:20,123 num_classes: 10
2021-10-13 00:25:20,124 epochs: 60
2021-10-13 00:25:20,135 grad_clip: 5.0
2021-10-13 00:25:20,135 log_frequency: 100
2021-10-13 00:25:20,135 model: {'name': 'ResNet18', 'num_classes': 10}
2021-10-13 00:25:20,136 criterion: {'name': 'CrossEntropyLoss'}
2021-10-13 00:25:20,136 optimizer: {'name': 'SGD', 'lr': 0.1, 'weight_decay': 0.0005, 'momentum': 0.9}
2021-10-13 00:25:20,136 scheduler: {'name': 'CosineAnnealingLR', 'T_max': 60, 'eta_min': 0.0}
2021-10-13 00:25:20,136 dataset: {'name': 'DatasetGenerator', 'train_batch_size': 128, 'eval_batch_size': 128}
2021-10-13 00:25:20,138 seed: 0
2021-10-13 00:25:20,139 version: resnet18
2021-10-13 00:25:20,139 exp_name: path/to/your/experiment/folder
2021-10-13 00:25:20,139 config_path: configs/cifar10
2021-10-13 00:25:20,140 load_model: False
2021-10-13 00:25:20,140 data_parallel: False
2021-10-13 00:25:20,140 train_batch_size: 512
2021-10-13 00:25:20,140 eval_batch_size: 512
2021-10-13 00:25:20,141 num_of_workers: 8
2021-10-13 00:25:20,141 train_data_type: CIFAR10
2021-10-13 00:25:20,141 train_data_path: ../datasets
2021-10-13 00:25:20,141 test_data_type: CIFAR10
2021-10-13 00:25:20,141 test_data_path: ../datasets
2021-10-13 00:25:20,142 universal_train_portion: 0.2
2021-10-13 00:25:20,142 universal_stop_error: 0.5
2021-10-13 00:25:20,142 universal_train_target: 'classwise'
2021-10-13 00:25:20,142 train_step: 10
2021-10-13 00:25:20,143 use_subset: False
2021-10-13 00:25:20,143 attack_type: min-min
2021-10-13 00:25:20,143 perturb_type: classwise
2021-10-13 00:25:20,143 patch_location: center
2021-10-13 00:25:20,144 noise_shape: [10, 3, 32, 32]
2021-10-13 00:25:20,144 epsilon: 0.03137254901960784
2021-10-13 00:25:20,144 num_steps: 1
2021-10-13 00:25:20,144 step_size: 0.006274509803921569
2021-10-13 00:25:20,145 random_start: False
2021-10-13 00:25:20,145 feature_dim: 128
2021-10-13 00:25:20,145 temperature: 0.5
2021-10-13 00:25:20,145 k: 200
2021-10-13 00:25:20,146 batch_size: 512
2021-10-13 00:25:20,146 epochs: 60
2021-10-13 00:25:20,146 arch: resnet18
2021-10-13 00:25:20,147 noise_num: 10
2021-10-13 00:25:20,147 save_image_num: 10
Training images:   0%|          | 0/98 [00:00<?, ?it/s]Training images:   0%|          | 0/98 [00:02<?, ?it/s]
Traceback (most recent call last):
  File "ssl_perturbation_save_model.py", line 622, in <module>
    main()
  File "ssl_perturbation_save_model.py", line 608, in main
    noise, save_name_pre = universal_perturbation(noise_generator, None, None, model, None, optimizer, None, random_noise, ENV, train_loader, train_noise_data_loader, batch_size, temperature, memory_loader, test_loader, k, train_data)
  File "ssl_perturbation_save_model.py", line 281, in universal_perturbation
    delta = torch.stack(class_noise_eta[key]).mean(dim=0).to(device) - random_noise[key]
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
